runtime/acala/src/lib.rs

```diff
@@ -1862,9 +1878,26 @@ pub type Executive = frame_executive::Executive<
 	frame_system::ChainContext<Runtime>,
 	Runtime,
 	AllPallets,
-	OnRuntimeUpgrade,
+	(OnRuntimeUpgrade, TransactionPaymentUpgrade),
 >;
 
+parameter_types! {
+	pub TokenFixedRates: Vec<(CurrencyId, Ratio, Balance)> = vec![
+		(DOT, calculate_asset_ratio(DotPerSecond::get(), AcaPerSecond::get()), FeePoolBootBalance::get()),
+		(AUSD, calculate_asset_ratio(AusdPerSecond::get(), AcaPerSecond::get()), FeePoolBootBalance::get()),
+	];
+}
+
+pub struct TransactionPaymentUpgrade;
+impl frame_support::traits::OnRuntimeUpgrade for TransactionPaymentUpgrade {
+	fn on_runtime_upgrade() -> Weight {
+		for asset in TokenFixedRates::get() {
+			let _ = <module_transaction_payment::Pallet<Runtime>>::initialize_pool(asset.0, asset.1, asset.2);
+		}
+		0
+	}
+}
```

Comment:
we should probably use a non-zero value for the Weight. We could benchmark it or just use the `max_weight` of a block

Reply:
Oh I get it, when runtime upgrade happens, we dont't expect other extrinsics in this block

Reply:
`on_runtime_upgrade` uses computation time (weight) just like anything else so we don't want to go over the .5 seconds of allocated time for block production. Using max block weight is just a shortcut so we don't have to benchmark anything, because no other normal extrinsic will be added to the block and therefore the block will be produced in time. In practice the function `initialize_pool` is computationally light enough that using zero for its weight would be fine as we have some leeway.